### PR TITLE
UIREC-17 Receiving: Access the Piece Details Wizard

### PR DESCRIPTION
**Pull request by AndreiShumski2**
## Purpose
Change Modal window with Item Details (Piece Details) to support checkboxes and make it consistent with the recent mockup
https://issues.folio.org/browse/UIREC-17

## Approach
Add checkboxes, logic for them, locations and item status, buttons `next` and `previous`, update Modal pane title

#### TODOS and Open Questions
Tests for `ItemDetails` component wil be written after updating `ReceivingList` component 

#### Screenshot
screenshot_3

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/components/Utils/getLocationsForSelect.js [hunk 1]
```diff
@@ -1,6 +1,4 @@
-import { get } from 'lodash';
-
-const getLocationsForSelect = (resources) => get(resources, 'locations.records', []).map(({ name, id, code }) => ({
```
Comment (aliaksei-chumakou) [inline src/components/Utils/getLocationsForSelect.js]: What is the purpose of this change? Within it you have to duplicate code of getting `_.get(resources, 'locations.records', [])` in every place where you want to use `getLocationsForSelect`

File: src/components/Receiving/ItemDetails.js [hunk 2]
```diff
@@ -6,62 +6,160 @@ import { get } from 'lodash';
...
+            ))}
+          </Select>
+        </div>
+      ),
     };
 
     return (
       <Modal
-        label={<FormattedMessage id="ui-orders.receiving.modalPaneTitle" />}
+        contentClass={css.paneTitle}
```
Comment (aliaksei-chumakou) [inline src/components/Receiving/ItemDetails.js]: This could be removed

File: src/components/Receiving/ItemDetails.js [hunk 3]
```diff
@@ -6,62 +6,160 @@ import { get } from 'lodash';
...
-          </Button>
-        </Row>
         <MultiColumnList
           contentData={itemList}
           formatter={resultsFormatter}
-          visibleColumns={['title', 'barcode', 'location', 'itemStatus']}
+          visibleColumns={['isChecked', 'barcode', 'format', 'location', 'itemStatus']}
           columnMapping={{
-            title: <FormattedMessage id="ui-orders.receiving.title" />,
+            isChecked: <Checkbox type="checkbox" checked={this.state.allChecked} onChange={() => this.toggleAll(itemList)} />,
```
Comment (aliaksei-chumakou) [inline src/components/Receiving/ItemDetails.js]: `itemList` comes from props, no need to pass it into `toggleAll` function, as well as no need to bind callback with arrow function here. It could be `onChange={this.toggleAll}`

File: src/components/Receiving/ItemDetails.js [hunk 4]
```diff
@@ -6,62 +6,160 @@ import { get } from 'lodash';
...
+        buttonStyle="primary"
+        disabled={!this.state.checkedItems.length}
+      >
+        <FormattedMessage id="ui-orders.receiving.nextBtn" />
+      </Button>
+    </Row>
+  );
+
+  toggleAll = (itemList) => {
+    if (!this.state.allChecked) {
```
Comment (aliaksei-chumakou) [inline src/components/Receiving/ItemDetails.js]: I think this condition should be checked inside `this.setState()` callback, since only there you get safe current value of `allChecked`

File: src/components/Receiving/ItemDetails.js [hunk 5]
```diff
@@ -6,62 +6,160 @@ import { get } from 'lodash';
...
+    }
+  }
+
+  toggleItem = (item) => {
+    const isItemChecked = this.state.checkedItems.filter(el => el.id === item.id).length;
+
+    if (isItemChecked) {
+      const checkedItemList = this.state.checkedItems.filter(el => el.id !== item.id);
+
+      this.setState({ checkedItems: [...checkedItemList] });
```
Comment (aliaksei-chumakou) [inline src/components/Receiving/ItemDetails.js]: you're changing `checkedItems` in state with values from `checkedItems` in current state. This usually done in callback of `this.setState()` https://reactjs.org/docs/react-component.html#setstate

File: src/components/Receiving/ItemDetails.js [hunk 6]
```diff
@@ -6,62 +6,160 @@ import { get } from 'lodash';
...
+            fullWidth
+            placeholder=" "
+          />
+        </div>
+      ),
+      'itemStatus': () => (
+        <div className={css.editableFieldWrapper}>
+          <Select
+            placeholder=" "
+            defaultValue=""
```
Comment (aliaksei-chumakou) [inline src/components/Receiving/ItemDetails.js]: I'm not sure if `defaultValue` is needed here

File: src/components/Utils/getLocationsForSelect.js (hunk 1 above)
Comment (AndreiShumski2) [reply src/components/Utils/getLocationsForSelect.js]: Rolled back. I reused this function in `ReceivigList`

File: src/components/Receiving/ItemDetails.js [hunk 7]
```diff
@@ -6,62 +6,148 @@ import { get } from 'lodash';
...
+      >
+        <FormattedMessage id="ui-orders.receiving.nextBtn" />
+      </Button>
+    </Row>
+  );
+
+  toggleAll = () => {
+    const { itemList } = this.props;
+
+    this.setState(({ allChecked }) => ({
```
Comment (aliaksei-chumakou) [inline src/components/Receiving/ItemDetails.js]: I would add more code for better understanding,
```
this.setState((state, props) => {
  const allChecked = !state.allChecked;
  const checkedItems = allChecked ? props.itemList || [];
  return {
    allChecked,
    checkedItems,
  };
});
```

Comment (aliaksei-chumakou) [reply src/components/Receiving/ItemDetails.js]: IMO, it's more visible than `checkedItems: allChecked ? [] : itemList,` - where you set empty array if `allChecked` is truthy

File: src/components/Receiving/ItemDetails.css [hunk 8]
```diff
@@ -0,0 +1,3 @@
+.editableFieldWrapper {
+  margin-top: 14px;
```
Comment (aliaksei-chumakou) [inline src/components/Receiving/ItemDetails.css]: doesn't it look like a hack to you? Maybe it make sense to check `Layout` helpers https://ux.folio.org/storybook/?selectedKind=Layout&selectedStory=Available%20classes&full=0&addons=1&stories=1&panelRight=0&addonPanel=REACT_STORYBOOK%2Freadme%2Fpanel
or LayoutGrid https://ux.folio.org/storybook/?selectedKind=LayoutGrid&selectedStory=Basic%20Usage&full=0&addons=1&stories=1&panelRight=0&addonPanel=REACT_STORYBOOK%2Freadme%2Fpanel

File: src/components/Receiving/ItemDetails.js [hunk 9]
```diff
@@ -6,62 +6,151 @@ import { get } from 'lodash';
...
+      </Button>
+      <Button
+        buttonStyle="primary"
+        disabled
+      >
+        <FormattedMessage id="ui-orders.receiving.previousBtn" />
+      </Button>
+      <Button
+        buttonStyle="primary"
+        disabled={!this.state.checkedItems.length}
```
Comment (aliaksei-chumakou) [inline src/components/Receiving/ItemDetails.js]: you're not consistent at this function. You're passing `close` here from props, but make direct access to `this.state.checkedItems`.
If you ask me, Footer looks like a React SFC to me

File: src/components/Receiving/ItemDetails.css (hunk 8 above)
Comment (AndreiShumski2) [reply src/components/Receiving/ItemDetails.css]: Input(select) has default style `margin-bottom: 1rem`
